src/new_storage/handle/reader.rs

```diff
@@ -41,40 +38,36 @@ impl Reader {
             block_size,
             cache,
             backend,
-            access_keys: Mutex::new(HashSet::new()),
         }
     }
 
     /// Try fetch the block from `MemoryCache`.
-    fn fetch_block_from_cache(&self, block_id: u64) -> Option<Arc<RwLock<Block>>> {
+    fn fetch_block_from_cache(&self, block_id: u64, version: u64) -> Option<Arc<RwLock<Block>>> {
         let key = CacheKey {
             ino: self.ino,
             block_id,
+            // Check current block is invalid or not.
+            version,
```

Comment:
It might be better not change the cache key, put version into cache item instead.
when old version found from cache, the cache item can be deleted immediately.

Reply:
Ok, now that the structure of the Map is HashMap<K, Arc<RwLock<Block>>>, try to associate the version with the block and remove the version field from the cache key, but it may slightly affect the retrieval speed.